src/tools/time_governor.cc

```diff
@@ -55,7 +55,17 @@ const Tuple & TimeGovernor::get_input_time_type(double lower_bound, double upper
 
 
 const Record & TimeGovernor::get_input_type() {
-	return Record("TimeGovernor",
+    static const Tuple &dt_step =
+        Tuple("DtLimits", "Time dependent changes in min_dt and max_dt limits.")
+            .declare_key("time", TimeGovernor::get_input_time_type(), Default::obligatory(),
+                    "The start time of dt step set.")
+            .declare_key("min_dt", TimeGovernor::get_input_time_type(), Default::read_time("'min_dt' value of TimeGovernor."),
+                    "Soft lower limit for the time step.")
+            .declare_key("max_dt", TimeGovernor::get_input_time_type(), Default::read_time("'max_dt' value of TimeGovernor."),
+                    "Whole time of the simulation if specified, infinity else.")
+            .close();
+
+    return Record("TimeGovernor",
             "Setting of the simulation time. (can be specific to one equation)")
```

Comment:
Improve doc of the TimeGovernor setting to provide wider picture. Describe which keys have preference in limiting time steps and brief description of 'fixed TimeMarks' and where they may come from.